Cherry-picked commits for WordPress 6.4 RC1 v3

SiobhyB opened the pull request:
Latest commits for WordPress 6.4 RC1.

Included changes:

* [Image: Reimplement lightbox trigger as a minimal button in corner of image](https://github.com/WordPress/gutenberg/pull/55212#top)
* [Only suppress admin notices when JavaScript is enabled](https://github.com/WordPress/gutenberg/pull/55403)

dmsnell committed:
Unconditionally load the Gutenberg copy of the HTML Tag Processor (#55404)

Temporary fix to resolve breaking tests in `trunk` after the WordPress 6.4 backport.

SiobhyB committed:
Revert "Unconditionally load the Gutenberg copy of the HTML Tag Processor (#55404)"

This reverts commit 271587a46c2102e5e394326fd8205c3cd6e30167.

artemiomorales committed:
Image: Reimplement lightbox trigger as a minimal button in corner of image (#55212)

* Reimplement lightbox trigger as a minimal button in corner of image

* Remove obsolete directives

* Update directives to fire logic properly via image or button click

* Ensure lightbox button only appears when hovering over image, not whole figure

* Ensure close button does not receive focus when opening lightbox via mouse

* Ensure button only receives focus when lightbox is closed via keyboard

* Add comments

* Prevent unnecessary focus being shown on mobile

* Add dynamic positioning for button when image uses 'contain' setting

* WORK IN PROGRESS - Begin accounting for various edge cases

We need to account for the fact that an image can have
custom dimensions, aspect ratio, cover or contain, captions,
thumbnail dimensions, and potentially other scenarios.

This commit begins to address those issues but notably fails
in cases where one uses a horizontal image, at full scale,
with custom aspect ratio, using 'contain'.

It seems to work in all other cases that I've checked but
needs more thorough testing and the code can probably be cleaner,
and may contain some unnecessary items.

* Simplify and improve button placement logic

* Simplify logic to show button on hover

* Fix styles

* Simplify calls to showLightbox

* Fix style inconsistency between browsers

* Change button position slightly

* Reduce button offset

* Add style override for better consistency across themes

* Fix logic so lightbox animates as intended; remove extraneous code

* Update comment

afercia commented:
Noting that although https://github.com/WordPress/gutenberg/pull/55212 has been merged, the Lightbox still needs code and accessibility fixes.

peterwilsoncc committed:
[Edit Widgets] Only suppress admin notices when JS enabled. (#55403)

Only suppresses the display of notices on the widget block editor screen if JavaScript is available.

SiobhyB commented:
Merged considering time constraints to avoid delaying RC1 and with the following in mind: 

* Both of the cherry-picked changes have been reviewed/approved separate to this PR. 
* All tests passed, and these changes will be manually tested further before committing to core and as part of RC1.
* I spoke with @afercia, and the bugs he surfaced will be worked on separately for inclusion in RC2.
